spec/system/jobseekers_can_create_a_job_alert_from_a_search_spec.rb

```diff
@@ -0,0 +1,163 @@
+
+        context "when jobseeker is signed in" do
+          before { login_as(jobseeker, scope: :jobseeker) }
+
+          it "renders a link to job alerts dashboard" do
+            click_on I18n.t("subscriptions.confirm.view_job_alerts")
+            expect(current_path).to eq(jobseekers_subscriptions_path)
+          end
+        end
+
+        context "when jobseeker is signed out" do
+          it "renders a link to job alerts dashboard" do
+            click_on I18n.t("subscriptions.confirm.view_job_alerts")
+            expect(current_path).to eq(new_jobseeker_session_path)
+            sign_in_jobseeker
+            expect(current_path).to eq(jobseekers_subscriptions_path)
+          end
+        end
+      end
+    end
+  end
+
+  context "when a location category search is carried out" do
+    let(:location_category_search?) { true }
+    let(:location) { "London" }
+    let!(:location_polygon) { create(:location_polygon, name: "london") }
+
+    scenario "can successfuly subscribe to a new alert" do
+      visit jobs_path
+      and_perform_a_search
+      and_click_job_alert_link
+
+      expect(page).to have_content(I18n.t("subscriptions.new.title"))
+      and_the_search_criteria_are_populated
+
+      click_on I18n.t("buttons.subscribe")
+      expect(page).to have_content("There is a problem")
+
+      fill_in_subscription_fields
+
+      message_delivery = instance_double(ActionMailer::MessageDelivery)
+      expect(SubscriptionMailer).to receive(:confirmation) { message_delivery }
+      expect(message_delivery).to receive(:deliver_later)
+      click_on I18n.t("buttons.subscribe")
+
+      activities = PublicActivity::Activity.all
+      keys = activities.pluck(:key)
+      expect(keys).to include("subscription.alert.new")
+      expect(keys).to include("subscription.daily_alert.create")
+    end
+  end
+
+  context "when a location search is carried out" do
+    let(:location_category_search?) { false }
+    let(:location) { "SW1A 1AA" }
+
+    scenario "can successfuly subscribe to a new alert" do
+      visit jobs_path
+      and_perform_a_search
+      and_click_job_alert_link
+
+      expect(page).to have_content(I18n.t("subscriptions.new.title"))
+      and_the_search_criteria_are_populated
+
+      click_on I18n.t("buttons.subscribe")
+      expect(page).to have_content("There is a problem")
+
+      fill_in_subscription_fields
+
+      message_delivery = instance_double(ActionMailer::MessageDelivery)
+      expect(SubscriptionMailer).to receive(:confirmation) { message_delivery }
+      expect(message_delivery).to receive(:deliver_later)
+      click_on I18n.t("buttons.subscribe")
+
+      activities = PublicActivity::Activity.all
+      keys = activities.pluck(:key)
+      expect(keys).to include("subscription.alert.new")
+      expect(keys).to include("subscription.daily_alert.create")
+    end
+  end
+
+  def and_click_job_alert_link
+    if page.has_css?("#job-alert-link")
+      click_on("Receive a job alert")
```

Comment:
Should we be consistent and use I18n keys here too?

Reply:
I did think about it, but also didn't want to change too much things I was just copying across. In specs I find sometimes there is a balance to be had between something that is easy to read and understand (the string itself that I click on) and consistency and not repeating ourselves (using the i18n translation). Let me smash my head about it for a bit 😂

Reply:
I did it (the translation needed updating anyway to something a bit cleaner)